scripts/generate_serialization.py

```diff
@@ -0,0 +1,248 @@
+import os
```

Comment:
I would eventually add a CI job (as part of nightly tests) that could check that
`python3 scripts/generate_serialization.py` don't introduces any changes, to be sure it's aligned with the source code.

Reply:
That's a great idea and should probably be done for all auto-generated code (perhaps including the parser?). Out of scope for this PR though

Reply:
Have actually added this as part of the PR after all (also for the generate_functions)